XDA-3 Fix page to restore event emails

StephenCWills opened the pull request:
The custom Razor engine adds a `NodeHost` property to the view bag for Razor views. The view checks permissions and then invokes the new static method in `EventEmailNode`. The static method sends a web request to `/Node/EventEmail/RestoreEventEmails`, which forwards the request along to all instances of `EventEmailNode` in the cluster.

clackner-gpa reviewed (changes requested):
Per our discussion:
The Razor page should move into Notification Pages.
The API Endpoint will then be called by the Notification Pages Server side and use APIAuth

StephenCWills commented:
I made some changes over here to make the URL configurable. I left the `RestoreEventEmail.cshtml` page in because it doesn't hurt anything. We can remove it if you want.

I also added the `IsTripped` endpoint to the `EventEmailNode` web controller because previously that information wasn't available.

Lastly, I added an API controller to https://github.com/GridProtectionAlliance/SystemCenter/pull/239 to make use of these API calls. The only thing left is to implement the page to actually restore event emails.
